packages/create-bison-app/template/config.ts

```diff
@@ -0,0 +1,93 @@
+import { z } from 'zod';
+
+import { notEmpty } from './lib/type-witchcraft';
+
+const stages = ['production', 'development', 'test'] as const;
+
+type Stage = (typeof stages)[number];
+
+function getStage(stages: Stage[]) {
+  if (!stages.length) return 'development';
+
+  for (const stage of stages) {
+    // if any of the provided stages is not production, assume we aren't in production
+    if (stage !== 'production') {
+      return stage;
+    }
+  }
+
+  return stages[0];
+}
+
+function isStage(potentialStage: string): potentialStage is Stage {
+  return stages.includes(potentialStage as Stage);
+}
+
+function envToBoolean(value: string | undefined, defaultValue = false): boolean {
+  if (value === undefined || value === '') {
+    return defaultValue;
+  }
+
+  // Explicitly test for true instead of false because we don't want to turn
+  // something on by accident.
+  return ['1', 'true'].includes(value.trim().toLowerCase()) ? true : false;
+}
+
+export function isProduction() {
+  return stage === 'production';
+}
+
+export function isDevelopment() {
+  return stage === 'development';
+}
+
+export function isTesting() {
+  return stage === 'test';
+}
+
+export function isLocal() {
+  return isDevelopment() || isTesting();
+}
+
+// a bit more versatile form of boolean coercion than zod provides
+const coerceBoolean = z
+  .string()
+  .optional()
+  .transform((value) => envToBoolean(value))
+  .pipe(z.boolean());
+
+const configSchema = z.object({
+  stage: z.enum(stages),
+  ci: z.object({
+    isCi: coerceBoolean,
+    isPullRequest: coerceBoolean,
+  }),
+  database: z.object({
+    url: z.string(),
+    shouldMigrate: coerceBoolean,
```

Comment:
Do we want to default `shouldMigrate` to `true` or `process.env.NODE_ENV === 'production'`? I think this might trip people up.

Reply:
It's set explicitly by the `SHOULD_MIGRATE` environment variable. Which, I realized I forgot to add to the .env files.